### ISSUE TITLE
Possible null pointer dereference in OpcUa_String_Initialize()

### ISSUE THREAD

**Issue by cacamille3** [OPCFoundation/UA-AnsiC#80]
Currently the argument `OpcUa_String* a_pString` is not checked with
`OpcUa_ReturnErrorIfArgumentNull(a_pString); `

This could result to a segmentation fault in OPC UA Client or Server using this function which is available through _./core/opcua_string.h_

```
OpcUa_StatusCode OpcUa_String_Initialize( /*  bi */ OpcUa_String* a_pString )
{
    OpcUa_pStringInternal   pStringInt  = (OpcUa_pStringInternal)   a_pString;

    pStringInt->uMagic          = OpcUa_uiMagic;
    pStringInt->bFreeSecondMem  = OpcUa_True;
    /*pStringInt->uReserved       = 0;*/
    pStringInt->uLength         = 0;
    pStringInt->strContent      = OpcUa_Null;

    return OpcUa_Good;
}
```

**Comment by cacamille3** [comment]
Pull Request #81

**Comment by bernd-edlinger** [comment]
Actually I am not sure if OpcUa_String_Initialize should better return Void,
like all other OpcUa_&lt;DataType&gt;_Initialize functions.

Right now the "error tolerance" of the OpcUa_String objects makes more
problems than it helps.

**Comment by cacamille3** [comment]
What do you mean with "error tolerance" ? 
But yes it might be a better idea to follow the pattern of other Builtin types

**Comment by bernd-edlinger** [comment]
> What do you mean with "error tolerance" ?

I mean if an OpcUa_String_Initialize has been forgotten, the following
OpcUa_String_AttachXxx or OpcUa_String_Clear should crash immediately,
but they rarely do.

I for one would prefer if they aborted every time rather than silently ignoring
the problem until a certain bit pattern is on the uninitialized object.

**Comment by cacamille3** [comment]
Yes totally agree, I have notified this scenario in my OPC UA Server as well...

**Comment by bernd-edlinger** [comment]
Fixed by f4d0d18.
